src/ILCompiler/src/Program.cs

```diff
@@ -499,7 +518,7 @@ private int Run(string[] args)
                     _metadataLogFileName,
                     stackTracePolicy,
                     invokeThunkGenerationPolicy,
-                    new Internal.IL.CoreRTILProvider(),
+                    ilProvider ?? new CoreRTILProvider(),
```

Comment:
Nit: Just do `ILProvider ilProvider = new CoreRTILProvider()` above?

Reply:
That would break ReadyToRun.

Reply:
`ILProvider ilProvider = _isReadyToRunCodeGen ? new ReadyToRunILProvider() : new CoreRTILProvider()` ?

It does not feel right to create IL provider in multiple different places.